src/main/scala/mesosphere/marathon/api/v2/json/Formats.scala

```diff
@@ -349,7 +349,8 @@ trait ContainerFormats {
           }
         case _ =>
           if (`type` == ContainerInfo.Type.DOCKER) {
-            throw SerializationFailedException("docker must not be empty")
+            //TODO(janisz): Hack!!! Think about explicit way to validate exclusive container types.
```

Comment:
That was my first attempt (after trying to fix it with Format validation). [`JsResultException`](https://www.playframework.com/documentation/2.5.x/api/scala/index.html#play.api.libs.json.JsResultException) has constructor that will require much more code becouse it was not meant to use outside of the parsing framework. With this approach we reuse validation from `DockerContainerParameters` so it it get changed (e.g. new required field added) no more changes in code will be required.